SCons/Tool/compilation_db.xml

```diff
@@ -0,0 +1,130 @@
+<?xml version="1.0"?>
+<!--
+__COPYRIGHT__
+
+This file is processed by the bin/SConsDoc.py module.
+See its __doc__ string for a discussion of the format.
+-->
+
+<!DOCTYPE sconsdoc [
+        <!ENTITY % scons SYSTEM '../../doc/scons.mod'>
+        %scons;
+        <!ENTITY % builders-mod SYSTEM '../../doc/generated/builders.mod'>
+        %builders-mod;
+        <!ENTITY % functions-mod SYSTEM '../../doc/generated/functions.mod'>
+        %functions-mod;
+        <!ENTITY % tools-mod SYSTEM '../../doc/generated/tools.mod'>
+        %tools-mod;
+        <!ENTITY % variables-mod SYSTEM '../../doc/generated/variables.mod'>
+        %variables-mod;
+        ]>
+
+<sconsdoc xmlns="http://www.scons.org/dbxsd/v1.0"
+          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+          xsi:schemaLocation="http://www.scons.org/dbxsd/v1.0 http://www.scons.org/dbxsd/v1.0/scons.xsd">
+
+    <tool name="compilation_db">
+        <summary>
+            <para>
+                Sets up &b-CompilationDatabase; builder which generates a clang tooling compatible compilation database.
+            </para>
+        </summary>
+        <sets>
+            <item>COMPILATIONDB_COMSTR</item>
+            <item>__COMPILATIONDB_UACTION</item>
+            <item>__COMPILATIONDB_UTARGET</item>
+            <item>__COMPILATIONDB_USOURCE</item>
+            <item>__COMPILATIONDB_ENV</item>
+            <item>COMPILATIONDB_USE_ABSPATH</item>
+        </sets>
+    </tool>
+
+    <builder name="CompilationDatabase">
+        <summary>
+            <para>
+                The &b-CompilationDatabase; builder writes a
+                <ulink url="https://clang.llvm.org/docs/JSONCompilationDatabase.html">clang formatted compilation
+                    database
+                </ulink> which is consumed by a number of clang tools, editors, and other tools.
+            </para>
+            <para>
+                If you don't specify any files, the builder will default to <filename>compile_commands.json</filename>.
```

Comment:
This must be where you had the question about supplying args to a builder spec, which our tooling doesn't provide for.  Still, this sounds awkward: "if you don't speciffy any files"? The **Builder Objects** section in the manpage talks about calling builders, but doesn't seem to correlate with this usage?

Reply:
See the emitter..